cmd/gitops-server/cmd/cmd.go

```diff
@@ -90,6 +92,9 @@ func NewCommand() *cobra.Command {
 	cmd.Flags().StringVar(&options.OIDC.RedirectURL, "oidc-redirect-url", "", "The OAuth2 redirect URL")
 	cmd.Flags().DurationVar(&options.OIDC.TokenDuration, "oidc-token-duration", time.Hour, "The duration of the ID token. It should be set in the format: number + time unit (s,m,h) e.g., 20m")
 
+	cmd.Flags().BoolVar(&options.DevMode, "dev-mode", true, "Enables development mode")
+	cmd.Flags().StringVar(&options.DevUser, "dev-user", v1alpha1.DefaultClaimsSubject, "Sets development User")
```

Comment:
i think we can probably do without letting people set this and just have the dev user be the default

Reply:
The intent here is to allow us to have multiple users we can use to check behaviors. Now we have the admin one, but shortly we'll get the limited one, we might also want to have a middle term semi-limited one for instance.

Reply:
there are actually no real users at all, they are all just role binding(https://github.com/weaveworks/weave-gitops/blob/main/charts/gitops-server/templates/rolebinding.yaml#L18), so it's just easier to have another set of role and role binding in place deployed, via a `dev` chart, than override the current one.

Reply:
that role binding you linked is for the wego-server, did you mean https://github.com/weaveworks/weave-gitops/blob/main/charts/gitops-server/templates/admin-user.yaml#L16 ?

Reply:
yep that one.

Reply:
either way, the k8s user that the claims subject is bound to does not change, that username doesn't turn into anything concrete, so we can't really have more than one user at the same time

Reply:
users can set [this](https://github.com/weaveworks/weave-gitops/blob/main/charts/gitops-server/templates/admin-user.yaml#L56) but not [this](https://github.com/weaveworks/weave-gitops/blob/main/charts/gitops-server/templates/admin-user.yaml#L16), so there can only ever be one user at a time

Reply:
It can change, that's why I added the dev user, it's being set here: https://github.com/weaveworks/weave-gitops/pull/1855/files#diff-f77070871fc35c299f82f2108c7fe3c52e8a113e0fa100fdc03f353d07307770R73

So when making the calls, it will have that as impersonated user. So we can have as many roles/rolesbinding to test different permitions.

Reply:
ah so we are sort of implementing static users for this mode?

Reply:
no really, because we don't have any concept of password and user fetching, we just hardcode the user there and no matter the username and password sent, it will return the `dev-user` and the authenticated one.

As the name implies is for development only.